handler.go

```diff
@@ -89,6 +89,55 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	h.mux.ServeHTTP(w, r)
 }
 
+// makeUserRequiredHandler takes a custom handler and returns a standard handler, ensuring that
+// the custom handler is only called with a authenticated user.
+func (h *Handler) makeUserRequiredHandler(fn func(http.ResponseWriter, *http.Request, *User)) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		q := r.URL.Query()
+		username, password := q.Get("u"), q.Get("p")
```

Comment:
We should be taking then as either query parameters or as basic auth
headers. From a usability perspective, most developers find it much easier
to construct a query string than to figure out how to make their HTTP
library put in basic auth headers.

On Sun, Jan 4, 2015 at 5:19 PM, Levi Cook <notifications@github.com> wrote:

> In handler.go
> <https://github.com/influxdb/influxdb/pull/1284#discussion-diff-22443256>:
>
> > @@ -89,6 +89,55 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
> >  	h.mux.ServeHTTP(w, r)
> >  }
> >
> > +// makeUserRequiredHandler takes a custom handler and returns a standard handler, ensuring that
> > +// the custom handler is only called with a authenticated user.
> > +func (h *Handler) makeUserRequiredHandler(fn func(http.ResponseWriter, *http.Request, *User)) http.HandlerFunc {
> > +	return func(w http.ResponseWriter, r *http.Request) {
> > +		q := r.URL.Query()
> > +		username, password := q.Get("u"), q.Get("p")
>
> Feels strange to pass credentials as query params. Did we consider and
> pass on http basic?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/1284/files#r22443256>.
>